src/uct/ib/cm/cm_iface.c

```diff
@@ -56,6 +56,7 @@ static unsigned uct_cm_iface_progress(void *arg)
         ++count;
     }
 
+    /* we are in the progress() context. Now it is safe to release resources. */
```

Comment:
why is it safe to release resources now? what could happen if the resources would be released immediately?